Gjoranv/new metrics proxy 3

gjoranv opened the pull request:
Added one commit (Remove 'healthport') since last attempt.

gjoranv committed:
New metrics-proxy

gjoranv committed:
Use string constant from metrics-proxy to avoid duplication.

gjoranv committed:
Exclude http-utils and apache httpclient.

- Not needed to generate config, and saves 1 MB bundle size.

gjoranv committed:
Remove 'healthport' from vespa-services config.

- Only one port is used per service.

hmusum commented on metrics-proxy/src/main/resources/configdefinitions/vespa-services.def:
```diff
@@ -5,6 +5,5 @@ package=ai.vespa.metricsproxy.service
 service[].configId string
 service[].name string default=""
 service[].port int default=-1
-service[].healthport int default=-1
```
Removing fields in a config definition is not allowed.

bjorncs replied:
It should be okay as this is a new config definition introduced in this PR.

hmusum replied:
OK, sure, then it is OK

gjoranv committed:
Allow adding metrics to the default consumer, for model amenders.

gjoranv committed:
Move amending of default consumer to Admin.

- The metrics proxy cluster is not set up (because it is a
  per-node service) by the time the amender is run.

gjoranv committed:
Inline helper method.
